common/djangoapps/student/tests/test_create_account.py

```diff
@@ -561,34 +561,66 @@ def test_email_pattern_requirements(self, email, expect_success):
         else:
             self.assert_error(params, "email", "Unauthorized email address.")
 
-    ## Note andrey.lykhoman:  fix this test
     @ddt.data(
         ('false.result@sorry.bad', True, False),
         ('true.result@its.good', False, True),
     )
     @ddt.unpack
-    def test_email_uniqueness(self, email, is_avaliable, expect_success):
-
-
-        returning = MagicMock()
-        returning.exists = MagicMock(return_value=is_avaliable)
-        returning2 = MagicMock()
-        returning3 = MagicMock()
-        returning3.filter.return_value = returning
-        returning3.exclude.return_value = returning2
-        #returning.get.return_value = returning
-
-
-        with patch('django.contrib.auth.models.User.objects.exists', return_value=is_avaliable):
-            z = UserCreationFormExtended({'email': email})
-            errors = z.errors.as_data()
-            result = True
+    def test_email_uniqueness_create_form(self, email, is_avaliable, expect_success):
+        """
+        Testing uniqueness email in creation user in admin page.
+        You can't create user with certain email address if another user has this email address.
+        If you will try that you must have validation error.
+
+        :param email:
+        :param is_avaliable:
+        :param expect_success:
+        :return:
+        """
+        mockObject = MagicMock()
+        mockObject.exists.return_value = is_avaliable
```

Comment:
I would use a little bit different order like filter, exclude, and exists but it more like a nit.

Reply:
I rewrite code and use configure_mock method ( https://github.com/raccoongang/edx-platform/pull/327/files#diff-db937b02d16c28ef5ae9fff01fc1a72bR44)